Makefile

```diff
@@ -390,23 +388,28 @@ repository-test:
 
 # Main distribution
 
+check-repo: doc/CHANGES
```

Comment:
Please document what this does.  "check" is vague and easy to misinterpret.
Why does it depend on doc/CHANGES?  Would there be any harm in not depending on doc/CHANGES?
Also, is the result expected to be viewed by a human or to halt the process if it doesn't return a status of 0?  (I expect the latter.)